gear/gear/database.py

```diff
@@ -172,14 +165,17 @@ def __init__(self):
     def start(self, read_only=False):
         return TransactionAsyncContextManager(self.pool, read_only)
 
+    @retry_transient_mysql_errors
     async def just_execute(self, sql, args=None):
         async with self.start() as tx:
             await tx.just_execute(sql, args)
 
+    @retry_transient_mysql_errors
     async def execute_and_fetchone(self, sql, args=None):
         async with self.start() as tx:
             return await tx.execute_and_fetchone(sql, args)
 
+    @retry_transient_mysql_errors
```

Comment:
So I'm guessing from the code, we're not able to retry async generators. I think that's ok based on a quick look. But I think you can make it retry on `select_and_fetchone`. There's no async generator there.

Reply:
resolved